aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -80,15 +80,28 @@ static inline ScalarType combine_categories(ScalarType higher, ScalarType lower)
   // NOLINTNEXTLINE(bugprone-branch-clone)
   if(isComplexType(higher)) {
     return higher;
-  }
-  else if(!isComplexType(lower) && isFloatingType(higher)) {
+  } else if (isComplexType(lower)) {
+    // special case for bfloat16
+    if (higher == kBFloat16) {
+      // BFloat16 corresponds to Float in terms of range,
+      // so we return ComplexFloat
+      return kComplexFloat;
+    }
+    // preserve value type of higher if it is floating type.
+    if (isFloatingType(higher)) {
+      return toComplexType(higher);
```

Comment:
same as above, if higher if float and lower is chalf, the expected out dtype is chalf

Reply:
If I understand correctly, value type of `higher` (1D+ tensor incase of 1D+ tensor and scalar inputs) should preserve the value type of `higher`.

Also to verify I checked against the type_promotion utility in `_prims/util.py`.

```python
import torch
from torch._prims.utils import elementwise_dtypes, ELEMENTWISE_TYPE_PROMOTION_KIND

s = torch.tensor(1j, dtype=torch.chalf)  # lower
t = torch.zeros(3, dtype=torch.float)  # higher

print(elementwise_dtypes(s, t, type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT))

```

Output
```
(torch.complex64, torch.complex64) # (comp_dtype, result_dtype)
```